pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -394,22 +395,23 @@ def precision_plugin(self) -> PrecisionPlugin:
             self._precision_plugin = self.select_precision_plugin()
         return self._precision_plugin
 
-    @property
-    def training_type_plugin(self) -> TrainingTypePlugin:
-        if self._training_type_plugin_resolved:
-            # avoid calling `resolve_training_type_plugin` multiple times
-            return self._training_type_plugin
+    def final_training_type_plugin(self) -> TrainingTypePlugin:
         if self._training_type_plugin is None:
             self._training_type_plugin = self.select_training_type_plugin()
         self._training_type_plugin = self.resolve_training_type_plugin(self._training_type_plugin)
         # attach checkpoint plugin to the training type plugin
         if self._checkpoint_io is not None:
             self._training_type_plugin.checkpoint_io = self._checkpoint_io
-        precision_plugin = self.precision_plugin
-        if precision_plugin is not None:
-            self._training_type_plugin._precision_plugin = precision_plugin
-        self._training_type_plugin_resolved = True
-
+        if (
+            isinstance(self.strategy, TrainingTypePlugin) and self.strategy._precision_plugin is None
+        ) or not isinstance(self.strategy, TrainingTypePlugin):
+            precision_plugin = self.precision_plugin
+            if precision_plugin is not None:
+                self._training_type_plugin.precision_plugin = precision_plugin
+        if (isinstance(self.strategy, TrainingTypePlugin) and self.strategy.accelerator is None) or not isinstance(
+            self.strategy, TrainingTypePlugin
+        ):
+            self._training_type_plugin.accelerator = self.select_accelerator()
```

Comment:
@awaelchli this is really ugly, but this is the minimal impact way I can think about to handle custom plugins. How about we refactor/rewrite Accelerator_connector after this and keep logic as same as possible compare to before for now?

Reply:
Thank you. Yes whatever makes the tests pass. At this point I have no clue anymore.